catalog/transformer/catalog-transformer-tika-input/src/main/java/ddf/catalog/transformer/input/tika/TikaInputTransformer.java

```diff
@@ -235,6 +282,50 @@ public Metacard transform(InputStream input, String id)
         }
     }
 
+    private String getDatatype(String mimeType) {
+        if (mimeType == null) {
+            return null;
+        }
+
+        LOGGER.debug("Attempting to map {}", mimeType);
+        Optional<Map.Entry<com.google.common.net.MediaType, String>> returnType =
+                SPECIFIC_MIME_TYPE_DATA_TYPE_MAP.entrySet()
+                        .stream()
+                        .filter(mediaTypeStringEntry -> {
+                            com.google.common.net.MediaType mediaType =
```

Comment:
Reviewing this more, you're calling `parse(mimeType)` over and over again when its value isn't impacted by the iteration. You could call it once, before streaming through either collection.